### DISCUSSION TITLE
Lightstreamer's official Python library

### DISCUSSION THREAD

**Discussion by AlessandroAlinone** [ig-python/ig-markets-api-python-library#253] [General]
Just wanted to let you know that we are close to releasing the official Lightstreamer client library for Python. It will be maintained by the Lightstreamer team and will contain all the state-of-the-art features of the other Lightstreamer libraries (stream-sense, websockets, auto-reconnections, auto-resubscriptions, etc.).
The alpha should be available in the next few weeks. It will need to be unlocked on IG's servers, though.
Will keep you posted.

**Comment by AlessandroAlinone** [comment]
We have just released the Lightstreamer Python Client SDK 1.0.0-beta.1 on [PyPi](https://pypi.org/project/lightstreamer-client-lib/). It is a fully-featured client supporting websockets, stream-sensing, session recovery, auto-reconnections etc. It requires Python 3.9 and Lightstreamer Server 7.2.0.

In the README on PyPi there is a brief tutorial which shows how the client can connect to a Lightstreamer Server and subscribe to some items. If you have ever used another Lightstreamer client library, you will surely recognize the familiar API.

There is also [this demo](https://github.com/Lightstreamer/Lightstreamer-example-StockList-client-python) on Github, which subscribes to a list of randomly-generated stock items (not on IG but on a random simulator) and displays the formatted updates on the console.

The documentation is [here](https://lightstreamer.com/api/ls-python-client/1.0.0-beta.1/index.html). For the moment it uses a generic format, but we intend to switch to a more pythonic style in the future.

**NOTE**: The Lightstreamer Python Client SDK requires IG to upgrade to Lightstreamer Server v.7.2 or newer and unlock Python support in the license.

**Reply by bug-or-feature** [reply]
@AlessandroAlinone thanks for this. Are you communicating the need to upgrade to IG? Have they agreed to do it?

**Reply by AlessandroAlinone** [reply]
We have just notified them. We are waiting for a response.

**Reply by bug-or-feature** [reply]
Great - I'd be interested to hear how they respond

**Reply by starsnpixel** [reply]
@AlessandroAlinone any news on this matter?

**Reply by AlessandroAlinone** [reply]
*BIG NEWS*
IG upgraded to Lightstreamer Server v.7.3.3. This means that the official Lightstreamer Client library is now supported.
We have updated the IG page to reflect this: https://labs.ig.com/lightstreamer-downloads

**Reply by bug-or-feature** [reply]
@AlessandroAlinone this is great news, thanks for updating. I'll create a ticket to add support in this library. It would mean replacing module [lightstreamer.py](https://github.com/ig-python/trading-ig/blob/master/trading_ig/lightstreamer.py), is that correct?

**Reply by AlessandroAlinone** [reply]
Yes, the new library can be obtained via pip as explained here: https://pypi.org/project/lightstreamer-client-lib/1.0.3/
Pay attention that the library API is quite different from the current lightstreamer.py module.

**Reply by bug-or-feature** [reply]
Great thanks. Luckily, in my own project I now need to make use of the Streaming API. So its good timing 😄

**Reply by AlessandroAlinone** [reply]
Please remember to stick to version 1.0.3 of the library (compatible with Lightstreamer Server 7.3.3), as version 2.0.0 requires Lightstreamer Server 7.4.0 (not yet deployed by IG). The good news is that future upgrades from 1.0.3 to 2.0.0 will be seamless.

**Reply by bug-or-feature** [reply]
OK noted

**Reply by bug-or-feature** [reply]
Latest release adds support for the official lib, see [here](https://github.com/ig-python/trading-ig/releases/tag/0.0.21)
